cyral/resource_cyral_access_rules.go

```diff
@@ -0,0 +1,329 @@
+			if len(confList) > 0 {
+				config := confList[0].(map[string]interface{})
+				policyIDs := config["policy_ids"].([]interface{})
+				ids := make([]string, 0, len(policyIDs))
+				for _, policyID := range policyIDs {
+					ids = append(ids, policyID.(string))
+				}
+				accessRule.Config = &AccessRulesConfig{
+					AuthorizationPolicyInstanceIDs: ids,
+				}
+			}
+		}
+
+		accessRules = append(accessRules, accessRule)
+	}
+
+	arr.AccessRules = accessRules
+	return nil
+}
+
+var ReadRepositoryAccessRulesConfig = ResourceOperationConfig{
+	Name:       "RepositoryAccessRulesRead",
+	HttpMethod: http.MethodGet,
+	CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+		return fmt.Sprintf("https://%s/v1/repos/%s/userAccounts/%s/accessRules",
+			c.ControlPlane,
+			d.Get("repository_id").(string),
+			d.Get("user_account_id").(string),
+		)
+	},
+	NewResponseData: func(_ *schema.ResourceData) ResponseData {
+		return &AccessRulesResponse{}
+	},
+}
+
+func resourceRepositoryAccessRules() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manage access rules",
+		CreateContext: CreateResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryAccessRulesCreate",
+				HttpMethod: http.MethodPost,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					repoID := d.Get("repository_id").(string)
+					userAccountID := d.Get("user_account_id").(string)
+					d.SetId(
+						marshalComposedID(
+							[]string{
+								repoID,
+								userAccountID,
+							},
+							"/",
+						),
+					)
+					return fmt.Sprintf("https://%s/v1/repos/%s/userAccounts/%s/accessRules",
+						c.ControlPlane,
+						repoID,
+						userAccountID,
+					)
+				},
+				NewResourceData: func() ResourceData {
+					return &AccessRulesResource{}
+				},
+				NewResponseData: func(_ *schema.ResourceData) ResponseData {
+					return &AccessRulesResponse{}
+				},
+			},
+			ReadRepositoryAccessRulesConfig,
+		),
+		ReadContext: ReadResource(ReadRepositoryAccessRulesConfig),
+		UpdateContext: UpdateResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryAccessRulesUpdate",
+				HttpMethod: http.MethodPut,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/repos/%s/userAccounts/%s/accessRules",
+						c.ControlPlane,
+						d.Get("repository_id").(string),
+						d.Get("user_account_id").(string),
+					)
+				},
+				NewResourceData: func() ResourceData {
+					return &AccessRulesResource{}
+				},
+				NewResponseData: func(_ *schema.ResourceData) ResponseData {
+					return &AccessRulesResponse{}
+				},
+			},
+			ReadRepositoryAccessRulesConfig,
+		),
+		DeleteContext: DeleteResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryAccessRulesDelete",
+				HttpMethod: http.MethodDelete,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+
+					idPieces, err := unmarshalComposedID(d.Id(), "/", 2)
```

Comment:
Instead of having to unmarshal the composed ID, maybe we could use the d.Get("repository_id").(string) and d.Get("user_account_id").(string) instead. I think this would simplify the code, but the way it is also works just fine.

Reply:
I feel like this is slightly cleaner, as we use the marshal function to compute the ID and then the unmarshal function to read it, so if it's OK then I'll just keep this as-is